cmd/erasure-metadata.go

```diff
@@ -428,7 +428,13 @@ func listObjectParities(partsMetadata []FileInfo, errs []error) (parities []int)
 			parities[index] = -1
 			continue
 		}
-		parities[index] = metadata.Erasure.ParityBlocks
+		if !metadata.IsValid() {
+			parities[index] = -1
+			continue
+		}
+		if !metadata.Deleted {
```

Comment:
I am not sure I understand. metadata.Erasure.ParityBlocks is zero for delete markers, since a delete marker is not erasure coded. 

```
{
  "Versions": [
    {
      "Header": {
        "Flags": 0,
        "ModTime": "2023-01-12T14:41:26.733323+01:00",
        "Signature": "e7cf29c0",
        "Type": 2,
        "VersionID": "d349d6825c424edba8d6fae8f6aa04e5"
      },
      "Idx": 0,
      "Metadata": {
        "DelObj": {
          "ID": "00nWglxCTtuo1vro9qoE5Q==",
          "MTime": 1673530886733323000,
          "MetaSys": {}
        },
        "Type": 2,
        "v": 1673524235
      }
    }
  ]
}
```

I guess metadata.IsValid() is what is making the difference

Reply:
Still it's important to communicate what we want parity blocks to be honored for.